### PR TITLE
feat(gear-common,gtest): Introduce proper mailbox for `gtest` (part 1)

### PR DESCRIPTION
**Pull request by techraed**
Some changes needed to port mailbox from `gear-common` to `gtest`:
- Introduce `AuxiliaryMailbox` for non-wasm (std) env.
- change blocks data management, so block data is accessible as it is on pallet (no context global storage). That's needed to implement properly mailbox callbacks.

### CODE REVIEW

**Comment by techraed** [comment]
to be merged after #3996

File: common/src/auxiliary/mod.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        K2: Ord,
+    {
+        self.inner.get(key1).and_then(|map| map.get(key2))
+    }
+
+    /// Inserts a value under provided keys in the map.
+    pub fn insert(&mut self, key1: K1, key2: K2, value: V) -> Option<V>
+    where
+        K1: Ord,
+        K2: Ord + Clone,
```
Comment (techraed) [inline common/src/auxiliary/mod.rs:68]: remove `Clone`

File: gtest/src/blocks.rs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        self.move_blocks_by(1)
+    }
+
+    /// Adjusts blocks info by moving blocks by `amount`.
+    pub(crate) fn move_blocks_by(&self, amount: u32) -> BlockInfo {
+        BLOCK_INFO_STORAGE.with_borrow_mut(|block_info| {
+            let Some(block_info) = block_info.as_mut() else {
+                panic!("must initialized in a `BlocksManager::new`");
+            };
+            block_info.height = block_info.height.saturating_add(amount);
```
Comment (techraed) [inline gtest/src/blocks.rs:74]: remove `saturating_*` logic as `gtest` is run with debug asserts.

File: gtest/src/manager.rs [hunk 3]
```diff
@@ -1180,7 +1177,7 @@ impl JournalHandler for ExtManager {
         // the queue.
         self.gas_allowance = Gas(GAS_ALLOWANCE);
         self.dispatches.push_front(dispatch);
-        self.block_info.height = self.block_info.height.saturating_add(1);
+        self.blocks_manager.move_blocks_by(1);
```
Comment (techraed) [inline gtest/src/manager.rs:1180]:
```suggestion
        self.blocks_manager.next_block();
```

File: gtest/src/blocks.rs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    pub(crate) fn get(&self) -> BlockInfo {
+        BLOCK_INFO_STORAGE.with_borrow(|cell| {
+            cell.as_ref()
+                .copied()
+                .expect("must be initialized in a `BlocksManager::new`")
+        })
+    }
+
+    /// Move blocks by one.
+    pub(crate) fn new_block(&self) -> BlockInfo {
```
Comment (techraed) [inline gtest/src/blocks.rs:64]:
```suggestion
    pub(crate) fn next_block(&self) -> BlockInfo {
```

Comment (techraed) [reply gtest/src/blocks.rs]: done in [a10c45e](https://github.com/gear-tech/gear/pull/3999/commits/a10c45e0e5685cf2131f2a99fed219d55dabf4fb)

File: gtest/src/manager.rs (hunk 3 above)
Comment (techraed) [reply gtest/src/manager.rs]: done in [a10c45e](https://github.com/gear-tech/gear/pull/3999/commits/a10c45e0e5685cf2131f2a99fed219d55dabf4fb)

File: gtest/src/blocks.rs (hunk 2 above)
Comment (techraed) [reply gtest/src/blocks.rs]: done in [a10c45e](https://github.com/gear-tech/gear/pull/3999/commits/a10c45e0e5685cf2131f2a99fed219d55dabf4fb)

File: common/src/auxiliary/mod.rs (hunk 1 above)
Comment (techraed) [reply common/src/auxiliary/mod.rs]: done in [a10c45e](https://github.com/gear-tech/gear/pull/3999/commits/a10c45e0e5685cf2131f2a99fed219d55dabf4fb)

**Review by breathx — APPROVED** [review]
🔥

**Comment by breathx** [comment]
@techraed managing merging order is on you

**Pull request merged**